### PR TITLE
Added/Modified some methods in Pokemon.java

### PR DESCRIPTION
**Pull request by gionata-bisciari**
Changes made:
+Added canEvolve():boolean
*Modified canPowerUp():boolean (now it checks if the player has enough star dust to powerup)

### CODE REVIEW

**Comment by AbandonedCart** [comment]
This may be a partial overlap of https://github.com/Grover-c13/PokeGOAPI-Java/pull/588

**Comment by gionata-bisciari** [comment]
It's based on the latest commit, and I think that canPowerUp() shouldn't throw an InvalidCurrencyException  because it gets the currency name directly from Currency class, maybe I'm wrong but I think that

**Comment by AbandonedCart** [comment]
It's typically bad practice to ignore an exception and it passes along any issue with the currency from the server (not just the variable definition), but both PRs are based on the latest commit.

**Comment by FabianTerhorst** [comment]
The method should throw the exceptions, because we don´t ignore exceptions.

File: library/src/main/java/com/pokegoapi/api/pokemon/Pokemon.java [hunk 1]
```diff
@@ -172,7 +174,30 @@ public Result transferPokemon() throws LoginFailedException, RemoteServerExcepti
 	 * @throws RemoteServerException the remote server exception
 	 */
 	public boolean canPowerUp() throws LoginFailedException, RemoteServerException {
-		return getCandy() >= getCandyCostsForPowerup();
+		boolean can = false;
```
Comment (FabianTerhorst) [inline library/src/main/java/com/pokegoapi/api/pokemon/Pokemon.java:177]: remove the changes to the canPowerUp method and this could be merged.

**Comment by gionata-bisciari** [comment]
So, I'll close it.

**Pull request closed without merging**